iocage/lib/Jail.py

```diff
@@ -0,0 +1,500 @@
+            f"allow.chflags={self.config.allow_chflags}",
+            f"allow.mount={self.config.allow_mount}",
+            f"allow.mount.devfs={self.config.allow_mount_devfs}",
+            f"allow.mount.nullfs={self.config.allow_mount_nullfs}",
+            f"allow.mount.procfs={self.config.allow_mount_procfs}",
+            f"allow.mount.zfs={self.config.allow_mount_zfs}",
+            f"allow.quotas={self.config.allow_quotas}",
+            f"allow.socket_af={self.config.allow_socket_af}",
+            f"exec.prestart={self.config.exec_prestart}",
+            f"exec.poststart={self.config.exec_poststart}",
+            f"exec.prestop={self.config.exec_prestop}",
+            f"exec.stop={self.config.exec_stop}",
+            f"exec.clean={self.config.exec_clean}",
+            f"exec.timeout={self.config.exec_timeout}",
+            f"stop.timeout={self.config.stop_timeout}",
+            f"mount.fstab={self.path}/fstab",
+            f"mount.devfs={self.config.mount_devfs}"
+        ]
+
+        if self.host.userland_version > 9.3:
+            command += [
+                f"mount.fdescfs={self.config.mount_fdescfs}",
+                f"allow.mount.tmpfs={self.config.allow_mount_tmpfs}"
+            ]
+
+        command += [
+            "allow.dying",
+            f"exec.consolelog={self.logfile_path}",
+            "persist"
+        ]
+
+        humanreadable_name = self.humanreadable_name
+        try:
+            helpers.exec(command, logger=self.logger)
+            self.update_jail_state()
+            self.logger.verbose(
+                f"Jail '{humanreadable_name}' started with JID {self.jid}",
+                jail=self
+            )
+        except subprocess.CalledProcessError as exc:
+            code = exc.returncode
+            self.logger.error(
+                f"Jail '{humanreadable_name}' failed with exit code {code}",
+                jail=self
+            )
+            self.logger.verbose(exc.output, jail=self)
+            raise
+
+    def start_vimage_network(self):
+
+        self.logger.log("Starting VNET/VIMAGE", jail=self)
+
+        nics = self.config.interfaces
+        for nic in nics:
+
+            bridges = list(self.config.interfaces[nic])
+
+            try:
+                ipv4_addresses = self.config.ip4_addr[nic]
+            except:
+                ipv4_addresses = []
+
+            try:
+                ipv6_addresses = self.config.ip6_addr[nic]
+            except:
+                ipv6_addresses = []
+
+            net = Network.Network(
+                jail=self,
+                nic=nic,
+                ipv4_addresses=ipv4_addresses,
+                ipv6_addresses=ipv6_addresses,
+                bridges=bridges,
+                logger=self.logger
+            )
+            net.setup()
+            self.networks.append(net)
+
+    def stop_vimage_network(self):
+        for network in self.networks:
+            network.teardown()
```

Comment:
Just as a suggestion it would be interesting to see if we could teardown these networks in parallel, just a thought